src/edu/ucsb/cs56/projects/games/tetris/TetrisBoard.java

```diff
@@ -53,22 +91,359 @@ public TetrisBoard() {
+
+    	public void actionPerformed(ActionEvent e) {
+
+    		if(e.getSource() == RulesButton)
+		   	{
+		   		
+		   		if(!isPaused) pause();
+		   		String text;
+		   		text = "	RULES\n\n\nThis game is very similar\nto the classic game of tetris.\n\n" +
+		   		"The Controls are as Follows:\n\n" +
+		   		"Left Arrow: Move Block Left\n" +
+		   		"Right Arrow: Move Block Right\n" +
+		   		"Up Arrow: Rotate Block\n" + 
+		   		"Down Arrow: Soft Drop\n" +
+		   		"Space Bar: Hard Drop\n" + 
+		   		"p: Pause Game\n" + 
+		   		"\n\nHave Fun !";
+		   		textArea = new JTextArea(text);
+		   		textArea.setEditable(false);
+		   		window.add(textArea);
+		   		window.revalidate();
+		   		tetrisPanel.setVisible(false);
+		   		textArea.setVisible(true);
+		   		//textArea.toFront();
+
+		   		UnpauseButton.setVisible(true);
+		   		RestartButton.setVisible(false);
+		   		PauseButton.setVisible(false);
+		   		RulesButton.setVisible(false);
+		   		
+		   	}
+		   	else if(e.getSource() == UnpauseButton)
+		   	{
+		   		pause();
+		   		UnpauseButton.setVisible(false);
+		   		RestartButton.setVisible(true);
+		   		PauseButton.setVisible(true);
+		   		RulesButton.setVisible(true);
+		   		window.remove(textArea);
+		   		tetrisPanel.setVisible(true);
+		   		tetrisPanel.requestFocus();
+
+		   	} 
+		   	else if(e.getSource() == PauseButton)
+		   	{
+		   		pause();
+		   		
+		   	}
+		   	else if (e.getSource() == MusicButton){ 
+
+		   		playMusic();
+
+
+		   	}
+		   	else if (e.getSource() == RestartButton) {
+				restartGame();
+			}
+
+
+    	}
+
+    }
+
+
+
+    public void playMusic() {
+/*
+    	InputStream in = new FileInputStream("tetrisSong.mp3");
+
+    	AudioStream as = new AudioStream(in);
+
+    	AudioPlayer.player.start(as);
+*/
+
+
+
+
+
+
+
+
+/*
+    	String bip = "tetrisSong.mp3";
+    	Media hit = new Media(bip);
+    	MediaPlayer mediaPlayer = new MediaPlayer(hit);
+    	mediaPlayer.play();
+*/
+
+
+    	try{
+    		//File song = new File("~/home/mnaito/Downloads/tetrisSong.wav");
+
+    		//File song = new File(
+    		//URL url = this.getClass().getClassLoader().getResource("tetrisSong.mp3");
+            //AudioInputStream aStream = AudioSystem.getAudioInputStream(url);
+
+    		//AudioInputStream aStream = AudioSystem.getAudioInputStream(this.getClass().getResource("./cs/student/marshallnaito/cs56/cs56-games-tetris/src/edu/ucsb/cs56/projects/games/tetris/tetrisSong.wav"));
+    		//AudioInputStream aStream = AudioSystem.getAudioInputStream("/cs/student/marshallnaito/cs56/cs56-games-tetris/src/edu/ucsb/cs56/projects/games/tetris/tetrisSong.mp3");
+    		//AudioInputStream aStream = AudioSystem.getAudioInputStream(("/tetrisSong.wav"));
+    		//AudioInputStream aStream = AudioSystem.getAudioInputStream(song);
+    		AudioInputStream aStream = AudioSystem.getAudioInputStream(new File("X:/cs/student/marshallnaito/cs56/cs56-games-tetris/src/edu/ucsb/cs56/projects/games/tetris/tetrisSong.wav").getAbsoluteFile());
```

Comment:
I was unable to try out the music feature, since the path to the audio file seems to be a hard coded path that will only work on Marshall's computer